src/utils/dns-utils.ts

```diff
@@ -0,0 +1,327 @@
+import dns from "node:dns/promises"
+
+import { err, errAsync, ok, okAsync, Result, ResultAsync } from "neverthrow"
+
+import {
+  BUILT_WITH_ISOMER_LOGO,
+  DNS_CNAME_SUFFIXES,
+  DNS_INDIRECTION_DOMAIN,
+  DNS_KEYCDN_SUFFIX,
+  REDIRECTION_SERVER_IPS,
+} from "@root/constants"
+import MonitoringError from "@root/errors/MonitoringError"
+import logger from "@root/logger/logger"
+import { gb } from "@root/middleware/featureFlag"
+
+import { getMonitoringConfig } from "./growthbook-utils"
+
+export function checkCname(domain: string) {
+  return ResultAsync.fromPromise(dns.resolveCname(domain), () => {
+    logger.error({
+      message: "Error resolving CNAME",
+      meta: { domain, method: "checkCname" },
+    })
+    return new Error()
+  })
+    .andThen((cname) => {
+      if (!cname || cname.length === 0) {
+        return okAsync(null)
+      }
+
+      //! todo: remove, this is some colima weird logic
+      if (cname.length === 1 && cname[0] === domain) {
+        return okAsync(null)
+      }
```

Comment:
this seems like a todo? should this be here

Reply:
ps, sorry this has been removed